src/new/response.ts

```diff
@@ -0,0 +1,28 @@
+interface BaseResponse {
+  status: number;
+  body?: unknown;
+  headers?: Record<string, unknown>;
+}
+
+interface ErrorResponse extends BaseResponse {
+  status: 400 | 404 | 500;
+  body: {
+    message: string;
+    errors?: Record<string, string>[];
+  };
+}
+
+interface SuccessResponseWithBody<Body> extends BaseResponse {
+  status: 200;
+  body: Body;
+}
+
+interface SuccessResponseWithoutBody extends BaseResponse {
```

Comment:
I removed this because after checking the CFY API there were a lot of cases where we have a response object with a body for 201 and 204. You can use the normal with body and pass void as a generic if you are not expecting a response. Like here: https://github.com/carforyou/carforyou-listings-web/pull/4256/files#diff-6f25325f85c581f21ee89e971ea36df2ca994219b876209b802d4e52f1bdd640R24